text/0097-log-data-model.md

```diff
@@ -0,0 +1,1025 @@
+  conventions for key names and possible values that allow all parties that work
+  with the field to have the same interpretation of the data. See references to
+  semantic conventions for `Resource` and `Attributes` fields and examples in
+  [Appendix A](#appendix-a-example-mappings).
+
+The reasons for having these 2 kinds of fields are:
+
+- Ability to efficiently represent named top-level fields, which are almost
+  always present (e.g. when using encodings like Protocol Buffers where fields
+  are enumerated but not named on the wire).
+  
+- Ability to enforce types of named fields, which is very useful for compiled
+  languages with type checks.
+
+- Flexibility to represent less frequent data via key/value pair lists. This
+  includes well-known data that has standardized semantics as well as arbitrary
+  custom data that the application may want to include in the logs.
+
+When designing this data model we followed the following reasoning to make a
+decision about when to use use a top-level named field:
+
+- The field needs to be either mandatory for all records or be frequently
+  present in well-known log and event formats (such as `Timestamp`) or is
+  expected to be often present in log records in upcoming logging systems (such
+  as `TraceId`).
+
+- The field’s semantics must be the same for all known log and event formats and
+  can be mapped directly and unambiguously to this data model.
+  
+Both of the above conditions were required to give the field a place in the
+top-level structure of the record.
+
+## Log and Event Record Definition
+
+Note: below we use type `any`, which can be a scalar value (number, string or
+boolean), or an array or map of values. Arbitrary deep nesting of values for
+arrays and maps is allowed (essentially allow to represent an equivalent of a
+JSON object).
+
+[Appendix A](#appendix-a-example-mappings) contains many examples that show how
+existing log formats map to the fields defined below. If there are questions
+about the meaning of the field reviewing the examples may be helpful.
+
+Here is the list of fields in a log record:
+
+Field Name     |Description
+---------------|--------------------------------------------
+Timestamp      |Time when the event occurred.
+TraceId        |Request trace id.
+SpanId         |Request span id.
+TraceFlags     |W3C trace flag.
+SeverityText   |The severity text (also known as log level).
+SeverityNumber |Numerical value of the severity.
+ShortName      |Short event identifier.
+Body           |The body of the log record.
+Resource       |Describes the source of the log.
+Attributes     |Additional information about the event.
+
+Below is the detailed description of each field.
+
+### Field: `Timestamp`
+
+Type: Timestamp, uint64 nanoseconds since Unix epoch.
+
+Description: Time when the event occurred measured by the origin clock. This
+field is optional, it may be missing the timestamp is unknown.
+
+### Trace Context Fields
```

Comment:
Does the trace context need to be elevated to top-level attribute? Can't it be just part of attributes?

Reply:
Yes, it can work either way. The rationale for making certain fields top-level instead of making that part of attributes is in the "Field Kinds" section.